tsinfer/formats.py

```diff
@@ -2405,11 +2405,16 @@ def sites_ancestral_allele(self):
             except IndexError:
                 unknown_alleles[allele] += 1
             ret[i] = allele_index
-        if sum(unknown_alleles.values()) > 0:
+        tot = sum(unknown_alleles.values())
+        if tot > 0:
             warnings.warn(
-                "The following alleles were not found in the variant_allele array "
-                "and will be treated as unknown:\n"
-                f"{unknown_alleles}"
+                "Ancestral alleles not found in the variant_allele "
```

Comment:
What'll happen here if there's thousands of them? The warning may take very long to generate, and may not be helpful. I think just keep it to the overall summary.

Reply:
The summary is the same order of length, right? As in it was, `{'aaa': 3, 'bbb':4}`, and would now be `'aaa': 3 (x%)\n'bbb': 4 (y%)`.

I'm suggesting this because I would find it useful to know (a) the proportion of bad ancestral alleles and (b) if I accidentally wasn't e.g. converting them to uppercase etc in the right way. I think reporting this would help avoid lots of accidentally bad analysis, which is why we introduced it (see https://github.com/tskit-dev/tsinfer/issues/870 and https://github.com/tskit-dev/tsinfer/issues/683#issuecomment-1163465893. But happy to be guided by user experience.

I've been testing on the default 1000 genomes VCFs, and seeing what would be useful.

Reply:
Ah, I didn't understand the code.

Can you do the computations outside of the string then? This is unnecessarily terse

Reply:
Refactored to move the calculations above the string construction.